keras_cv/layers/preprocessing/mix_up.py

```diff
@@ -0,0 +1,77 @@
+import tensorflow as tf
+import tensorflow.keras as keras
+from tensorflow.keras import layers
+
+
+class MixUp(layers.Layer):
+    """
+    MixUp implements the MixUp data augmentation technique as proposed in https://arxiv.org/abs/1710.09412.
+
+    Args:
+        num_classes: number of classes in the dataset.
+        alpha: alpha parameter for the sample distribution.
+        probability: probability to apply the CutMix augmentation.
+        label_smoothing: coefficient used in label smoothing.
+
+    Sample usage:
+    ```python
+    (x_train, y_train), (x_test, y_test) = keras.datasets.cifar10.load_data()
+    mixup = MixUp(10)
+    augmented_data, updated_labels = mixup(x_train, y_train)
+    ```
+    """
+    def __init__(
+        self,
+        num_classes,
+        alpha=0.8,
+        probability=1.0,
+        num_classes=None,
```

Comment:
this seems to be duplicated.

Reply:
ah good catch, fixed.  I actually managed to refactor it to not require num_classes at all.  Even better!